src/raftstore/store/worker/pd.rs

```diff
@@ -174,6 +182,57 @@ impl<T: PdClient> Runner<T> {
             error!("report split failed {:?}", e);
         }
     }
+
+    fn handle_validate_peer(&self, local_region: metapb::Region, peer: metapb::Peer) {
+        metric_incr!("pd.get_region");
+        match self.pd_client.get_region(local_region.get_start_key()) {
+            Ok(mut pd_region) => {
+                metric_incr!("pd.get_region");
+                if pd_region == local_region {
+                    // Local region info is as fresh as pd, which means
+                    // local peer is still in the region, let it be there.
+                    info!("peer {:?} is still valid because local region info is as fresh as pd",
+                          peer);
+                    return;
+                }
+                if !(pd_region.get_start_key() == local_region.get_start_key()) {
```

Comment:
If PD doesn't support a interface like `get_region_info_by_region_id`, it's impossible to check what change has been done in the region range and validate peer at this piece of code without scanning the region info all over.

A example:
When a node B of region range [100, 200) is isolated from the cluster, the region has been through a merge
```text
[0, 100) + [100, 200] -> [0,200)
```
and then a split
```text
[0, 200) -> [0, 100), [100, 200]
```

If node B is not removed out of the cluster by member change, at the end it would has the region range [0, 100), which doesn't overlap its original range [100, 200).
At this moment, ask PD to `get_region(by any key between [100, 200))` would get a incorrect region info. Actually we don't know what region node B is right now without a scan the whole table of key range indexing region info in PD.

As I discussed with @BusyJay at comments above, if we need to "do the right thing" in TiKV to validate stale peer correctly considering the incoming region merge feature, then PD would also need to "do the right thing" to provide the interface of "get region by region id", and PD should also be revised(probably change its region info indexing) to adapt the region merge feature.
So we'd better carry out the design of the region merge. It's not wise to make big changes to TiKV and PD to make it work with the incoming region merge feature without a overall picture of how region merge works.

cc @huachaohuang @qiuyesuifeng